cxx-qt-gen/src/parser/signals.rs

```diff
@@ -0,0 +1,145 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::parser::parameter::ParserParameter;
+use crate::syntax::fields::fields_named_to_ident_type;
+use syn::{Ident, ItemEnum, Result};
+
+/// Describes an individual Signal
+pub struct ParsedSignal {
+    /// The name of the signal
+    pub ident: Ident,
+    /// The parameters of the signal
+    pub parameters: Vec<ParserParameter>,
+    // TODO: later this will describe of the signal has an attribute
+    // stating that the signal exists in the base class
+}
+
+/// Describes a Signals block for a QObject
+pub struct ParsedSignals {
+    /// The name of the signals enum
+    pub ident: Ident,
+    /// The original enum for the signals
+    pub item: ItemEnum,
+    /// A list of the signals defined in the enum
+    pub signals: Vec<ParsedSignal>,
+}
+
+impl ParsedSignals {
+    /// Constructs a ParsedSignals object from a given [syn::ItemEnum] block
+    pub fn from(item: &ItemEnum, attr_index: usize) -> Result<Self> {
+        // Remove the attribute index as we have processed it
+        let mut item = item.clone();
+        item.attrs.remove(attr_index);
+
+        let signals = item
+            .variants
+            .iter()
+            .map(|variant| {
+                // Read the fields into parameter blocks
+                let parameters = fields_named_to_ident_type(&variant.fields)?
+                    .into_iter()
+                    .map(|(ident, ty)| ParserParameter { ident, ty })
+                    .collect();
+
+                Ok(ParsedSignal {
+                    // TODO: later we might have attributes on the signal
+                    // to state if they exist in the base class, these could be
+                    // extracted and stored in ParsedSignal here
+                    ident: variant.ident.clone(),
+                    parameters,
+                })
+            })
+            .collect::<Result<Vec<ParsedSignal>>>()?;
+
+        Ok(Self {
+            ident: item.ident.clone(),
+            item,
+            signals,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::tests::tokens_to_syn;
+    use quote::quote;
+    use syn::Type;
+
+    /// Helper which returns a f64 as a [syn::Type]
+    fn f64_type() -> Type {
+        tokens_to_syn(quote! { f64 })
+    }
+
+    #[test]
+    fn test_parsed_signals_from_empty() {
+        let e: ItemEnum = tokens_to_syn(quote! {
+            #[cxx_qt::signals(MyObject)]
+            enum MySignals {}
+        });
+        let signals = ParsedSignals::from(&e, 0);
+        assert!(signals.is_ok());
+        let signals = signals.unwrap();
```

Comment:
```suggestion
        let signals = ParsedSignals::from(&e, 0).unwrap();
```
It's okay to just unwrap, the test will just fail if it doesn't work.

Reply:
Same in the other test cases.

Reply:
Done